File: Messaging-App.Domain/DTOs/MessageGroupToReturnDto.cs
using System;

namespace Messaging_App.Domain.DTOs
{
    public class MessageGroupToReturnDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string LastSender { get; set; }
        public string LastMessage { get; set; }
        public DateTime? LastSent { get; set; }
        public bool IsGroup { get; set; }
    }
}

File: messaging-app-flutter/lib/dtos/message_group_to_return_dto.dart
class MessageGroupToReturnDto {
  final int id;
  final String name;
  final String lastSender;
  final String lastMessage;
  final String lastSent;
  final bool isGroup;

  MessageGroupToReturnDto(
    this.id,
    this.name,
    this.lastSender,
    this.lastMessage,
    this.lastSent,
    this.isGroup,
  );

  MessageGroupToReturnDto.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        lastSender = json['lastSender'],
        lastMessage = json['lastMessage'],
        lastSent = json['lastSent'],
        isGroup = json['isGroup'];
}
